archivy/data.py

```diff
@@ -193,6 +193,40 @@ def update_item(dataobj_id, new_content):
     load_hooks().on_edit(converted_dataobj)
 
 
+def update_item_frontmatter(dataobj_id, new_frontmatter):
+    """
+    Given an object id, this method overwrites the front matter
+    of the post with `new_frontmatter`.
+
+    ---
+    date: Str
+    id: Str
+    path: Str
+    tags: List[Str]
+    title: Str
+    type: note/bookmark
+    ---
+    """
+
+    from archivy.models import DataObj
+
+    filename = get_by_id(dataobj_id)
+    dataobj = frontmatter.load(filename)
+    for key in list(new_frontmatter):
+        dataobj[key] = new_frontmatter[key]
+    md = frontmatter.dumps(dataobj)
+    with open(filename, "w", encoding="utf-8") as f:
+        f.write(md)
+
+    # TODO: What does this do?
```

Comment:
This basically sends out a hook where the user can call code on the newly edited item (https://archivy.github.io/reference/hooks/).

I'll refactor this in another PR though